Monolithic crystal

granadogmarc opened the pull request:
Adding of a new digitizer GateVirtualSegmentationSD. The Virtual Segmentation Digitizer module provides a mechanism to generate an ID based on a virtual segmentation of the monolithic crystal, aligned with its spatial resolution. Ideally, the pitch size should be at least half of the position resolution. This virtual segmentation occurs post-simulation, ensuring that the simulation speed remains uncompromised even when dealing with large systems and numerous crystals such as total-body PET scans.

Commit:
Method UpdateVectorID(G4int new_copyNo) and addition of the GateDiscretizer.cc and GateDiscretizer.hh

Commit:
DummyDigitizerModule coppied into DiscretizerModule

Commit:
After adding the DiscretizerModule Files

Commit:
Addition of a new method for ChangeOutputID and new methods implemented in DiscretizerModule

Commit:
Clean up of the code, working for XYZ case

Commit:
Generalisation of the XYZ axis, bug for ID fixed, introduction of a dummy python script for macro_conversion

Commit:
After cleaning up the code and changing spatial resolution for pitch and adding macro generator flag

granadogmarc committed:
Update digitizer_and_detector_modeling.rst with Virtual segmentation

granadogmarc committed:
Update digitizer_and_detector_modeling.rst

granadogmarc committed:
Update digitizer_and_detector_modeling.rst

granadogmarc committed:
Update digitizer_and_detector_modeling.rst

granadogmarc committed:
Update digitizer_and_detector_modeling.rst

granadogmarc committed:
Update digitizer_and_detector_modeling.rst

granadogmarc committed:
Update digitizer_and_detector_modeling.rst

granadogmarc committed:
Update digitizer_and_detector_modeling.rst

granadogmarc committed:
Update digitizer_and_detector_modeling.rst

granadogmarc committed:
Update digitizer_and_detector_modeling.rst

granadogmarc committed:
Update digitizer_and_detector_modeling.rst

granadogmarc committed:
Update digitizer_and_detector_modeling.rst

granadogmarc committed:
Update digitizer_and_detector_modeling.rst

granadogmarc committed:
Update digitizer_and_detector_modeling.rst

granadogmarc committed:
Update digitizer_and_detector_modeling.rst

granadogmarc committed:
Update digitizer_and_detector_modeling.rst

Commit:
Clean up of the code, flexibility for the axis and command for the macro generator.

Commit:
Addition of the test for the needed free depths and comments

	modified:   source/digits_hits/src/GateVirtualSegmentationSD.cc

tontyoutoure commented:
Hi, may I ask why you need a virtual segmentation. Shouldn't some simple python scripting with np.histogram2d of the hit position be sufficient?

granadogmarc commented:
Hi Li Kun,

first of all, sorry for the late reply, I was on holiday when you sent your
email and this was deep down in my inbox and it took a while to get to it.

The idea behind this was to make it available for other people working with
monolithic crystals; and avoid the extra handling of big data files. We aim
to work with a total-body PET scanner and the amount of data will be
enormous. Thus, we thought this would be an optimised way to deal with this
issue.

Cheers,
M

El jue, 15 ago 2024 a las 8:04, LI Kun ***@***.***>)
escribió:

> Hi, may I ask why you need a virtual segmentation. Shouldn't some simple
> python scripting with np.histogram2d of the hit position be sufficient?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenGATE/Gate/pull/691#issuecomment-2290727759>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BJSAZTMHDYWWRVCY2GKIXYDZRRAGNAVCNFSM6AAAAABLMDXDO2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDEOJQG4ZDONZVHE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

Commit:
Cleanup GateSpatialResolution changes

	modified:   source/digits_hits/include/GateSpatialResolution.hh
	modified:   source/digits_hits/src/GateSpatialResolution.cc

Commit:
Remove .project and .cproject files

granadogmarc committed:
Update GateVirtualSegmentationSD.cc

granadogmarc committed:
Update GateVirtualSegmentationSD.cc

Commit:
fix bug with VirtualSegmentation for new SpatialResolution
